src/briefcase/config.py

```diff
@@ -33,11 +33,30 @@ def is_valid_pep508_name(app_name):
 
 def is_reserved_keyword(app_name):
     """Determine if the name is a reserved keyword."""
-    return keyword.iskeyword(app_name.lower()) or app_name.lower() in RESERVED_WORDS
+    return (
+        keyword.iskeyword(app_name.lower())
+        or app_name.lower() in RESERVED_WORDS
+        or keyword.iskeyword(app_name)
+    )
 
 
 def is_valid_app_name(app_name):
-    return not is_reserved_keyword(app_name) and is_valid_pep508_name(app_name)
+    """Determine if the app name is valid.
+
+    Checks the following:
+        - It is not a reserved keyword.
+        - It is a valid PEP508 name.
+
+    :param app_name: The app name to validate.
+    :returns: True if the app name is valid; False otherwise.
+    """
+    return all(
+        [
+            is_valid_pep508_name(app_name),
```

Comment:
As noted in my previous review - PEP508 seems like it's a historical red herring here. PEP508 imposes restrictions that don't apply to app names (no trailing `-` or `_`), and allows things that aren't valid (starting with digits and allowing `.`). 

AFAICT, what we need her is entirely determined by "is valid identifier after converting `-` to `_`", excluding reserved words.